### ISSUE TITLE
taup arrival times are wrong for custom velocity model

### ISSUE THREAD

**Issue by martijnende** [obspy/obspy#2816]
(I've seen numerous closed/open issues that report bugs with taup, but I did not immediately recognise my problem as being a duplicate of previously opened issues. Please forgive my possible duplicate report)

# Problem description

I've been playing around with taup models for solving hydro-acoustic problems. To this end I created a custom velocity model (`tvel` format) using IASPEI91 as a starting model. For some reason I get various error messages when removing redundant layers, but that's a problem for another day.

`acoustic_model.tvel`:

```
Modified IASPEI91 model
Anything below 2.75 km is bogus
     0.000    1.5062    0.001     1.0
     2.750    1.5493    0.001     1.0
     2.750    5.8000    3.3600    2.7200
    20.000    5.8000    3.3600    2.7200
    20.000    6.5000    3.7500    2.9200
    35.000    6.5000    3.7500    2.9200
    35.000    8.0400    4.4700    3.3198
```

When I run the following code, I get some results, but the arrival times (even the direct ones) are completely wrong:

```
import os
import obspy
from obspy.taup import TauPyModel
from obspy.taup.velocity_model import VelocityModel
from obspy.taup.taup_create import build_taup_model

filename = "acoustic_model.tvel"

vmodel = VelocityModel.read_tvel_file(filename)
assert vmodel.validate()

taup_model = build_taup_model(filename, output_folder=os.getcwd())

model = TauPyModel(model=f"{filename[:-5]}.npz")

arrivals = model.get_ray_paths(
    source_depth_in_km=2.,
    distance_in_degree=5,
)

print(arrivals[:2])

fig = plt.figure(figsize=(9, 5), constrained_layout=True)
ax = arrivals[:2].plot_rays(plot_type="cartesian", fig=fig)
```

Output:
```
2 arrivals
	p phase arrival at 2.349 seconds
	P phase arrival at 2.696 seconds
```

![taup](https://user-images.githubusercontent.com/6959003/112023689-c3d45e80-8b33-11eb-9241-8e26f50fce22.png)

While I find it curious that the diffracted arrival does not exactly hit the receiver, I am more worried about the direct phase which supposedly arrives after 2.3 seconds. The acoustic speed in water is about 1.5 km/s, so to bridge a 5 degree gap would take about 6 minutes (which I verified with an Eikonal simulation).

# Other information

ObsPy version: `1.2.2`
Installation method: `conda`
Operating system: `Linux Mint 19`

**Comment by ThomasLecocq** [comment]
euh, indeed, that looks strange !

**Comment by paitor** [comment]
Hi

The travel time is accurate if assuming a 35 km radii sphere (i.e. radii equal to the max depth in your velocity model). Presumably you need to add a point at the center of the earth (or whatever body you aim at, e.g. venus) in your velocity model.

**Comment by martijnende** [comment]
Thanks @paitor, adding a bottom layer that extends down to the centre of the planet results in much more reasonable travel times. When I explicitly set `vmodel.radius_of_planet = R` after defining `vmodel = VelocityModel.read_tvel_file(filename)` I do not manage to override the radius. It would be nice if this were an option, because otherwise `build_taup_model` needs to map down to the entire earth, which could take a long time. I also noticed there's a `layers` keyword in `VelocityModel` that is not documented in detail. Do you know how to use this? Since I only have 1 layer of interest (effectively), I might be able to define a velocity model as:

```
vmodel = VelocityModel(
    "acoust_model", 
    radius_of_planet=R, min_radius=R-35, max_radius=R, 
    moho_depth=100, cmb_depth=200, iocb_depth=300, is_spherical=True,
    layers=layers
)
```

**Comment by paitor** [comment]
Hi

Sorry, I'm not familiar with taup (nor the obspy implementation) as I've never used the later (confirmed my answer instead using my homebrewed traveltime calculator). Perhaps if you take a peak in the source code you'll be able to figure out how to define the input value on the layers parameter.

**Comment by ThomasLecocq** [comment]
@martijnende what I usually do is taking the AK135 model and change the top 30 km to my needs, this way you're sure the model is compatible

**Comment by megies** [comment]
@martijnende this is fixed by #3064 now in master
